include/api/CAnomalyJobConfig.h

```diff
@@ -12,16 +12,47 @@
 #include <api/ImportExport.h>
 
 #include <model/CLimits.h>
+#include <model/FunctionTypes.h>
 
 #include <rapidjson/document.h>
 
 #include <string>
 #include <vector>
 
+namespace CAnomalyJobLimitTest {
+struct testAccuracy;
+struct testLimit;
+struct testModelledEntityCountForFixedMemoryLimit;
+}
+
+namespace CAnomalyJobTest {
+struct testBadTimes;
+struct testOutOfSequence;
+struct testControlMessages;
+struct testSkipTimeControlMessage;
+struct testIsPersistenceNeeded;
+struct testModelPlot;
+struct testInterimResultEdgeCases;
+struct testRestoreFailsWithEmptyStream;
+}
+
+namespace CForecastRunnerTest {
+struct testSummaryCount;
+struct testRare;
+struct testInsufficientData;
+}
+
+namespace CStringStoreTest {
+struct testPersonStringPruning;
+struct testAttributeStringPruning;
+struct testInfluencerStringPruning;
+}
+
```

Comment:
It's not nice that so many tests need to be friends.

It seems that most, if not all, of these only need to be friends for the following pattern:

```
        ml::api::CAnomalyJobConfig jobConfig;
        jobConfig.analysisConfig().addDetector("a", "b", "c", "d", "e");
```

Therefore, I would create a `public` `static` method `ml::api::CAnomalyJobConfig CTestAnomalyJob::makeSimpleJobConfig(const std::string& functionName, const std::string& fieldName, const std::string& byFieldName, const std::string& overFieldName, const std::string& partitionFieldName, const TStrVec& influencers = {}, const std::string& summaryCountFieldName = "")` and make _that_ the sole friend of this class.  Then make all the tests switch to the pattern:

```
        ml::api::CAnomalyJobConfig jobConfig = CTestAnomalyJob::makeSimpleJobConfig("a", "b", "c", "d", "e");
```